cirq-core/cirq/ops/clifford_gate_test.py

```diff
@@ -530,3 +530,25 @@ def test_to_phased_xz_gate(trans_x, trans_z):
     assert np.isclose(
         actual_phased_xz_gate.axis_phase_exponent, expect_phased_xz_gates.axis_phase_exponent
     )
+
+
+def _extract_clifford_tableau_to_matrix(tableau):
+    matrix = np.zeros((tableau.n * 2, tableau.n * 2 + 1), dtype=np.int)
+    matrix[:, : tableau.n] = tableau.xs[: 2 * tableau.n]
+    matrix[:, tableau.n : 2 * tableau.n] = tableau.zs[: 2 * tableau.n]
+    matrix[:, tableau.n * 2] = tableau.rs[: 2 * tableau.n]
+    return matrix
+
+
+def test_clifford_tableau():
+    seen_tableau = []
+    for trans_x, trans_z in _all_rotation_pairs():
+        gate = cirq.SingleQubitCliffordGate.from_xz_map(trans_x, trans_z)
+        tableau_matrix = _extract_clifford_tableau_to_matrix(gate.clifford_tableau)
+        tableau_number = sum(2 ** i * t for i, t in enumerate(tableau_matrix.ravel()))
```

Comment:
This is nice but maybe we can simplify it further, e.g. the `_str_full_` might be cool too to use here as well? BTW - a digression - that `_str_full_` is something that I don't see being used anywhere else, I wonder why we wouldn't just use that as `_str_`?

Reply:
I do use `_str_full_`several times during the debugging. I just feel using string in test code is harder to manipulate compared with array. This code is a little bit out-of-date, actually. I have created `matrix` method in the tableau class. PTAL. But if you have better ideas, I'd love to update.